xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/hibernate/query/HqlQueryExecutor.java

```diff
@@ -103,32 +112,39 @@ public void initialize() throws InitializationException
         configuration.addInputStream(Util.getResourceAsStream(MAPPING_PATH));
     }
 
-    private Set<String> getAllowedNamedQueries()
+    private Set<String> getSafeNamedQueries()
     {
-        if (this.allowedNamedQueries == null) {
+        if (this.safeNamedQueries == null) {
             synchronized (this) {
-                if (this.allowedNamedQueries == null) {
-                    this.allowedNamedQueries = new HashSet<>();
+                if (this.safeNamedQueries == null) {
+                    this.safeNamedQueries = new HashSet<>();
 
                     // Gather the list of allowed named queries
                     Collection<NamedQueryDefinition> namedQueries =
                         this.hibernate.getConfigurationMetadata().getNamedQueryDefinitions();
                     for (NamedQueryDefinition definition : namedQueries) {
-                        if (HqlQueryUtils.isSafe(definition.getQuery())) {
-                            this.allowedNamedQueries.add(definition.getName());
+                        try {
+                            if (isSafe(definition.getQuery())) {
+                                this.safeNamedQueries.add(definition.getName());
+                            }
+                        } catch (QueryException e) {
+                            this.logger.warn("Failed to validate named query [{}] with statement [{}]: {}",
+                                definition.getName(), definition.getQuery(), ExceptionUtils.getRootCauseMessage(e));
                         }
                     }
                 }
             }
         }
 
-        return this.allowedNamedQueries;
+        return this.safeNamedQueries;
     }
 
     /**
      * @param statementString the statement to evaluate
      * @return true if the select is allowed for user without PR
+     * @deprecated
```

Comment:
Missing documentation.

Reply:
I don't really have any alternative to put there, actually. It's technically an API because it's protected, but I don't think it was really the intent.

Reply:
Wouldn't it make sense to point to the new `HQLStatementValidator`, even if it is internal? Or to make that interface non-internal?

Reply:
> Wouldn't it make sense to point to the new HQLStatementValidator, even if it is internal?

I'm afraid some people might actually use an internal tool without thinking about it if that's what we recommend (since that's what this javadoc is about).

> Or to make that interface non-internal?

I really don't want to start an infinite debate on if it's good enough to be public (I would prefer something more generic myself).